account_asset_management/wizard/account_asset_remove.py

```diff
@@ -1,188 +1,242 @@
-        asset_obj = self.pool.get('account.asset.asset')
-        asset = asset_obj.browse(cr, uid, context.get('active_id'))
-        if asset:
-            acc = asset.category_id.account_residual_value_id
-        return acc and acc.id or False
-
-    _columns = {
-        'date_remove': fields.date(
-            'Asset Removal Date', required=True,
-            help="Removal date must be after the last posted entry "
-                 "in case of early removal"),
-        'period_id': fields.many2one(
-            'account.period', 'Force Period',
-            domain=[('state', '<>', 'done')],
-            help="Keep empty to use the period of the removal ate."),
-        'sale_value': fields.float('Sale Value'),
-        'account_sale_id': fields.many2one(
-            'account.account', 'Asset Sale Account',
-            domain=[('type', '=', 'other')]),
-        'account_plus_value_id': fields.many2one(
-            'account.account', 'Plus-Value Account',
-            domain=[('type', '=', 'other')]),
-        'account_min_value_id': fields.many2one(
-            'account.account', 'Min-Value Account',
-            domain=[('type', '=', 'other')]),
-        'account_residual_value_id': fields.many2one(
-            'account.account', 'Residual Value Account',
-            domain=[('type', '=', 'other')]),
-        'posting_regime': fields.selection(
-            _posting_regime, 'Removal Entry Policy',
-            required=True,
-            help="Removal Entry Policy \n"
-                 "  * Residual Value: The non-depreciated value will be "
-                 "posted on the 'Residual Value Account' \n"
-                 "  * Gain/Loss on Sale: The Gain or Loss will be posted on "
-                 "the 'Plus-Value Account' or 'Min-Value Account' "),
-        'note': fields.text('Notes'),
-    }
-
-    _defaults = {
-        'sale_value': _get_sale_value,
-        'account_sale_id': _get_sale_account,
-        'account_plus_value_id': _get_plus_account,
-        'account_min_value_id': _get_min_account,
-        'account_residual_value_id': _get_residual_account,
-        'posting_regime': _get_posting_regime,
-    }
+    @api.model
+    def _default_account_plus_value_id(self):
+        asset_id = self._context.get('active_id')
+        asset = self.env['account.asset'].browse(asset_id)
+        return asset.profile_id.account_plus_value_id
+
+    @api.model
+    def _default_account_min_value_id(self):
+        asset_id = self._context.get('active_id')
+        asset = self.env['account.asset'].browse(asset_id)
+        return asset.profile_id.account_min_value_id
+
+    @api.model
+    def _default_account_residual_value_id(self):
+        asset_id = self._context.get('active_id')
+        asset = self.env['account.asset'].browse(asset_id)
+        return asset.profile_id.account_residual_value_id
+
+    @api.model
+    def _selection_posting_regime(self):
+        return[
+            ('residual_value', _('Residual Value')),
```

Comment:
I doubt that you should translate those here

Reply:
Just tested, and still needed :-1: